### PR TITLE
Add baselineversion metric

### PR DESCRIPTION
**Pull request by petrkotas**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/8573067/ 

### What this PR does / why we need it:

Introduce cluster baseline version metric. The metric signals whether the cluster lags behind
upgrade stream with its version.

Metric adds baselineVersion as additional dimension to the existing clusterVersion metric.
This dimension is emited only when the cluster requires attention: the version is behind its upgrade stream.

### Test plan for issue:
Added unit test covering baseline computation. Added unit test testing the metric emit.

### Is there any documentation that needs to be updated for this PR?
VSTS ticket https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/8573067/

### CODE REVIEW

File: pkg/monitor/cluster/clusterversions.go [hunk 1]
```diff
@@ -34,13 +34,20 @@ func (mon *Monitor) emitClusterVersions(ctx context.Context) error {
...
 		"desiredVersion":                       desiredVersion(cv),
 		"provisionedByResourceProviderVersion": mon.oc.Properties.ProvisionedBy, // last successful Put or Patch
 		"resourceProviderVersion":              version.GitCommit,               // RP version currently running
 		"operatorVersion":                      operatorVersion,                 // operator version in the cluster
-	})
+	}
+
+	baseline, upgrade := baselineVersion(cv, version.Streams)
+	if upgrade {
+		dims["baselineVersion"] = baseline
```
Comment (jim-minter) [inline pkg/monitor/cluster/clusterversions.go:47]: suggest we call this availableVersion

File: pkg/monitor/cluster/clusterversions.go [hunk 2]
```diff
@@ -65,3 +72,20 @@ func desiredVersion(cv *configv1.ClusterVersion) string {
 
 	return cv.Status.Desired.Version
 }
+
+// baselineVersion checks the upgradeStreams for possible upgrade of the cluster
+// when the upgrade is possible withing the current Y version, return true and closest upgrade stream version
+func baselineVersion(cv *configv1.ClusterVersion, streams []*version.Stream) (string, bool) {
```
Comment (jim-minter) [inline pkg/monitor/cluster/clusterversions.go:78]: can simplify this nicely by removing the bool return value - just return "" or the version.  Then the if check and `dims :=` change above is no longer needed.

**Review by jim-minter — COMMENTED** [review]
2 minor nits then good to go

File: pkg/monitor/cluster/clusterversions.go [hunk 3]
```diff
@@ -34,13 +34,16 @@ func (mon *Monitor) emitClusterVersions(ctx context.Context) error {
...
 		"actualVersion":                        actualVersion(cv),
 		"desiredVersion":                       desiredVersion(cv),
-		"provisionedByResourceProviderVersion": mon.oc.Properties.ProvisionedBy, // last successful Put or Patch
-		"resourceProviderVersion":              version.GitCommit,               // RP version currently running
-		"operatorVersion":                      operatorVersion,                 // operator version in the cluster
-	})
+		"provisionedByResourceProviderVersion": mon.oc.Properties.ProvisionedBy,      // last successful Put or Patch
+		"resourceProviderVersion":              version.GitCommit,                    // RP version currently running
+		"operatorVersion":                      operatorVersion,                      // operator version in the cluster
+		"availableVersion":                     baselineVersion(cv, version.Streams), // current available version for upgrade from stream
```
Comment (mjudeikis) [inline pkg/monitor/cluster/clusterversions.go:43]: rename function too please

File: pkg/monitor/cluster/clusterversions.go [hunk 4]
```diff
@@ -34,13 +34,16 @@ func (mon *Monitor) emitClusterVersions(ctx context.Context) error {
 		}
 	}
 
-	mon.emitGauge("cluster.versions", 1, map[string]string{
```
Comment (mjudeikis) [inline pkg/monitor/cluster/clusterversions.go:37]: And as we doign all in one revert this `dim`

**Review by mjudeikis — APPROVED** [review]

**Pull request merged**